File: leetcode_solutions/0026-remove-duplicates-from-sorted-array/solution.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 1;
        int iReal = 0;
        for (int iCurr = 0; iCurr < nums.length; iCurr++){
            if (nums[iCurr] != nums[iReal]){
                nums[iReal++ + 1] = nums[iCurr];
                k++;
            }
        }
       
        return k;
    }
}

File: leetcode_solutions/0026-remove-duplicates-from-sorted-array/solution.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int iReal = 0;
        int k = 1; // assume at least one unique integer
        for (int iCurr = 0; iCurr < nums.size(); iCurr++){
            int currNum = nums[iCurr];
            if (currNum != nums[iReal]){
                nums[iReal++ + 1] = currNum;
                k++;
            }
        }
        return k;

    }
};
